packages/runtime/container-runtime/src/blobManager.ts

```diff
@@ -424,8 +430,16 @@ export class BlobManager extends TypedEventEmitter<IBlobManagerEvents> {
 					this.deletePendingBlobMaybe(id);
 			  }
 			: undefined;
+
+		// If noLocalIds is true, return the storage id if it exists. This will ensure that
+		// clients running 1.x or older versions can read this blob because those versions didn't
+		// support local ids for blobs created in attached state.
+		let idForHandle = id;
+		if (this.noLocalIds) {
+			idForHandle = this.redirectTable.get(id) ?? id;
```

Comment:
I wonder, in general, what are the guarantees that local ID space and global ID space are not intersecting?
If they do, we can be up for surprise

Reply:
Local IDs are created in 2 scenarios:
1. Detached container - In this case, the local ID seems to be an integer starting with 0 and it keeps growing. I would think it is almost guaranteed that this will not intersect with storage ID. Right?
2. Attached container - In this case, the local ID is a uuid so it should be unique.